src/Coordination/Changelog.cpp

```diff
@@ -381,20 +387,20 @@ void Changelog::readChangelogAndInitWriter(uint64_t last_commited_log_index, uin
 
             LOG_TRACE(log, "Continue to write into {}", description.path);
             current_writer = std::make_unique<ChangelogWriter>(description.path, WriteMode::Append, description.from_log_index);
-            current_writer->setEntriesWritten(result.entries_read);
+            current_writer->setEntriesWritten(last_log_read_result.entries_read);
```

Comment:
Why we can be sure that `last_log_read_result` corresponds to the last element in `existing_changelogs`? What if we exited by some condition (probably error) in for-loop above?

Reply:
Thanks, good catch.